webapp/tests/elster_client/mock_erica.py

```diff
@@ -84,7 +84,7 @@ def mocked_elster_requests(*args, **kwargs):
                 response = MockErica.get_tax_offices()
             elif _PYERIC_API_BASE_URL_01 + '/tax_number_validity' in args[0]:
                 sub_urls = args[0].split('/')
-                response = MockErica.is_valid_tax_number(state_abbreviation=sub_urls[2], tax_number=sub_urls[3])
+                response = MockErica.is_valid_tax_number(state_abbreviation=sub_urls[len(sub_urls)-2], tax_number=sub_urls[len(sub_urls)-1])
```

Comment:
Can we get a Length < 0? why did you change that?

Reply:
I ran the acceptance tests with the mockEria and got problems. This makes sense because the erica_url is: http://127.0.0.1/v1

Sending a request for the tax number validity would then be send to http://127.0.0.1/v1/tax_number_validity/bw/1234.

Therefore, we get the following array with split: [http:,127.0.0.1, v1, tax_ number_validity, bw, 1234]

In the Tests the url is different. So, I did not want to change the hardcover values but make it dynamic. You are right, if the len is 0 or smaller than 2, we would get an error. However, we would have also gotten an error with the old values. Furthermore, we should not get a length of 0, because at least tax_number_validity, state_abbreviation and tax_number should be set. Even if they are not set, getting an error (while probably a more explicit one) is fine.

Reply:
Okay thank you for the explanation :)